client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,48 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy is set to {@link RetentionPolicy.RetentionType#CONSUMPTION},
+     * and the user expects reads from this Reader Group to impact data retention in the Stream,
+     * the retentionConfig in {@link ReaderGroupConfig} should be set to
+     * to 'CONSUMPTION_BASED_USER_STREAMCUT' or 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * Setting these options implies the Reader Group will notify the Controller of its consumption position {@link StreamCut}
+     * in the Stream, so only un-consumed data can be retained.
+     * This notification can be manual ('CONSUMPTION_BASED_USER_STREAMCUT') or automatic ('CONSUMPTION_BASED_AT_LAST_CHECKPOINT')
+     * If a Stream's Retention Policy is TIME/SPACE based the ReaderGroupConfig should have retentionConfig='NO_IMPACT',
+     * so consumption position notifications are not sent to Controller.
+     *
+     * NO_IMPACT - Set when read positions of this Reader Group are not expected to impact Stream truncation/retention.
```

Comment:
```suggestion
     * NO_IMPACT - Set when the reader's positions of this Reader Group should not impact Stream retention.
```

Reply:
Changed NO_IMPACT to NONE as suggested. fixed